### PR TITLE
set up associations

### PR DESCRIPTION
**Pull request by Jeanbulambo**
In this PR, We implemented the following elements:
- Created a class Classroom with the following:
@label instance variable should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
- Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Created the has-many side (a classroom has many students).
- Created the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classroom's students.
- Created a class Book with the following:
@title and @author instance variables should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
- Created a class Rental with the following:
@date instance variable should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Created the has-many side of Book and Rental (a book has many rentals).
- Created the belongs-to side of Rental and Book (a rental belongs to a book).
- Created the has-many side of Person and Rental (a person has many rentals).
- Created the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class ClassRoom
+  attr_accessor :label, :students
```
Comment (OLIPLICHE) [inline classroom.rb]: - Please declare students by using `attr_reader :students` because the method apply here is getter method

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    book.rentals << self
+
+    @person = person
+    person.rentals << self
+  end
+end
```
Comment (OLIPLICHE) [inline rental.rb:12]: - **NOTE: Kindly Make available and visible the missing files for better Review:**
# In student folder:
- **A student belongs to a classroom**:
Capture
- You need to add codes like this:
Capture
- [Open here to see an example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person)
+    Rental.new(date, self, person)
+  end
+end
```
Comment (OLIPLICHE) [inline book.rb:14]: - **NOTE: Kindly Make available and visible the missing files for better Review:**
# In Person file

Capture
- **@rentals should be initialized**: Add code like this `@rentals = []`
- **A method for adding rentals should be present**: Please [open here to see an example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Jeanbulambo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
Linter is green (the last commit have a green check mark (✅)) 👏
README file is descriptive 👏
Commit messages are meaningful 👏 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @Jeanbulambo,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🔥🔥💯💯



## Highlights ✨✨
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the functionality✅
📌 Good commit messages ✅


## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**